posydon/visualization/plot_pop.py

```diff
@@ -245,9 +246,11 @@ def plot_popsyn_over_grid_slice(pop, grid_type, met_Zsun, slices=None, channel=N
         met_indices = np.where(pop.select(where=channel_sel, columns=['metallicity']) == met_Zsun)[0].tolist()
         if 'HMS-HMS' in grid_path:
             slice_3D_var_range = (dq_edges[i],dq_edges[i+1])
-
-            sel = 'index in '+str(met_indices)+' & event == "ZAMS"'
-            data = pop.history.select(where=sel, columns=['S1_mass','S2_mass' ,'orbital_period'])
+            if len(met_indices) == 0:
+                data = pd.DataFrame(columns=['S1_mass','S2_mass' ,'orbital_period'])
+            else:
+                sel = 'index in '+str(met_indices)+' & event == "ZAMS"'
```

Comment:
How well is the select to deal with long statements? I was wondering, whether we shouldn't hard code the selection in the binary evolution but instead call the flow chart to extract all cases, where a binary will get send to a step with a binary grid?

Reply:
This is a good question. Especially with the other two selections, I'm not sure they're always correct. I have not changed the type of selection in this PR nor in the previous PR that touched this code.

Reply:
Please open an issue in case you don't want to address this here.